src/compressed_tensors/quantization/lifecycle/initialize.py

```diff
@@ -82,38 +84,52 @@ def initialize_module_for_quantization(
         _initialize_attn_scales(module)
 
     else:
+        if not isinstance(module, torch.nn.Linear):
+            _LOGGER.warning(f"Attempting to quantize module of type {type(module)}")
+
+        # use weight to determine observed shapes and dtype
+        if hasattr(module, "weight"):
+            weight = module.weight
+            assert isinstance(weight, torch.Tensor)
+        else:
+            # Note that a weight is required for both weight and activation
+            # quantization in order to know the dtype of activation scales
+            _LOGGER.warning(
+                f"module type {type(module)} targeted for quantization but "
+                f"has no attribute weight, skipping quantization for {type(module)}"
+            )
+            return
+
         if scheme.input_activations is not None:
-            _initialize_scale_zero_point(
+            initialize_qparams(
                 module,
                 "input",
                 scheme.input_activations,
+                observed_shape=(1, weight.shape[-1]),
```

Comment:
how can we reason about these `observed_shape` settings? the one for weight makes sense, but why 
- `observed_shape=(1, weight.shape[-1])` for inputs
- `observed_shape=weight.shape[:-1]` for outputs

Reply:
```
:param observed_shape: last (right-most) known dimensions of the observed weight/act
```

The first dim of input activations being 1 is probably a stretch, we can change to this just `weight.shape[-1]`